platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/ConcurrentArray.java

```diff
@@ -250,19 +258,32 @@ void add(T element) {
             throw new IllegalStateException("You can not call add on a immutable ConcurrentArray");
         }
 
-        // The sub-arrays in the dequeue all considered full except for the very last one. If the very last one is
+        // Lazily initialize the head of this copy.
+        if (head == null) {
+            synchronized (this) {
+                if (head == null) {
+                    SubArray<T> cur = current.get();
+                    if (cur.size.get() > 0) {
+                        cur.next = new SubArray<>(subarrayCapacity);
+                        cur = cur.next;
+                        current.set(cur);
+                    }
+                    head = cur;
+                }
+            }
+        }
+
+        // The sub-arrays in the list are all considered full except for the very last one. If the very last one is
         // also full, then add a new sub-array. We want to avoid locking for normal operations, and only
         // do so if the sub-array is full. So we will get the last sub-array, try to add to it, and if we fail
         // to do so, then we will acquire the lock to create a new sub-array.
-        SubArray<T> lastArray = arrays.getLast();
-        boolean success = lastArray.add(element);
+        boolean success = current.get().add(element);
         if (!success) {
             synchronized (this) {
```

Comment:
`synchronized (this) {` is the same as `synchronized {`.  Nitpick, but might make the code a touch easier to read.

Reply:
Odd, I distinctly remember the parenthetical expression being optional; it appears my recollection is flawed.